### DISCUSSION TITLE
API - list all payments

### DISCUSSION THREAD

**Discussion by Talej** [ACINQ/eclair#2443] [Developers]
The JSON API has /getsentinfo endpoint for retrieving details about a single payment. Is there an endpoint retrieving a list of payments to show history or when particular hash is not known?

**Comment by t-bast** [comment]
You can use the [audit API](https://acinq.github.io/eclair/#audit) for that, it lists incoming, relayed and outgoing payments that happened during a specific timeframe.
However, make sure you set the `to` and `from` parameter, otherwise this may return a lot of data!
We should probably add flags to filter what the caller is interested in (incoming, relayed and/or outgoing).

**Comment by Talej** [comment]
Thanks.

flags to filter would definitely be useful. Maybe also a flag to include failed payments as I think it only includes successful?
